contrib/babelfishpg_tds/src/backend/tds/tdslogin.c

```diff
@@ -1989,41 +1989,54 @@ TdsProcessLogin(Port *port, bool loadedSsl)
 	TdsErrorContext->phase = 0;
 	TdsErrorContext->reqType = TDS_LOGIN7;
 
-	PG_TRY();
+	/*
+	 * Check whether the first client message is indeed PRELOGIN.
+	 * 7.1 clients may send LOGIN7 instead, for example
+	 * jTDS sends PRELOGIN only when SSL is enabled.
+	 */
+	if (TdsReadNextBuffer() != EOF && TdsCheckMessageType(TDS_PRELOGIN))
```

Comment:
I am still concerned with this check. We are only checking message type to be PRELOGIN and proceeding ahead if it isn't assuming that first message would always be LOGIN request in case tds version = 7.1. There is no safety check validating that message is actually LOGIN request if it isn't PRELOGIN. We have [Assert here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/4217dbf92ee2c8f61a8840b090dcfd34d9114936/contrib/babelfishpg_tds/src/backend/tds/tdslogin.c#L763) but it may not work if engine is not configured with `--case-assert` flag. 

Can we check and throw an error upfront if `!TdsCheckMessageType(TDS_PRELOGIN) && !TdsCheckMessageType(TDS_LOGIN7)`

Reply:
Please note that the same scenario can tried today as well -  a valid prelogin followed by a corrupted login message.

We actually have the check in FetchLoginRequest, but as an assert.
```
Assert(TdsCheckMessageType(TDS_LOGIN7));
```
We should change this to an error message.

We're also validating the login request with all the length checks. So, it's unlikely to hit any issue even with corrupted packet.

Reply:
Thanks for the reviews! As I understand, the check for `LOGIN7` was just integrated a few hours ago in #2888. I've rebased the PR branch to the latest to re-run the tests.